### ISSUE TITLE
Batch size + logging

### ISSUE THREAD

**Issue by bnels** [DeepLearnPhysics/lartpc_mlreco3d#28]
We currently divide any metrics such as accuracy, (or other logged variables) by the batch size.  
This is due to the way that UResNets compute accuracy (by summing over the batch).

However, this makes computing metrics for GNNs troublesome, because accuracy is often computed over the batch graph (not a bunch of individual graphs).  Currently, we have to multiply all metrics by batch size, only to have it divided out later.

The problem is that somehow the GNN needs to know about batch size.  If there are events that have no em clusters (sometimes happens in Pi0 data), then this can't be determined only from the input and needs to be passed in somehow.  Unfortunately, the model/loss construction doesn't have access to the I/O config, so the batch size has to be listed again and passed explicitly to the model.

I think there are two solutions:
1) Have UResNet divide by batch size before returning, so we don't have to divide by batch size elsewhere
2) Make it easy to access batch size in I/O tools from a model

I think 1 is preferable above

**Comment by bnels** [comment]
Results of discussion:
1) in forward function of trainval:
default behavior should be:
if there is `count` in the name, we take sum
otherwise we take mean (don't explicitly divide by batch size)
2) logging:
will look for loss and accuracy keys.  If they exist, will print it out
3) loss should be called `loss` instead of `loss_seg`

All models should be updated to:
1) return mean over minibatch size
2) return `loss` instead of `loss_seg` for backpropagation

**Comment by francois-drielsma** [comment]
Changes listed by bnels have been implemented.

Models should no longer assume that what the dictionary returned by their loss function will be post-processed by the forward function in trainval. The forward function now takes the mean over number of GPUs by default and takes the sum of outputs if the keyword 'count' is in the variable name.

The output of the current models have been updated to fit this adjustment.

**Comment by bnels** [comment]
Closing thanks to #29
